elf.c

```diff
@@ -0,0 +1,49 @@
+/**
+ * @file elf.c
+ * @author Koen Van Bastelaere
+ * @brief Elf functions
+ * 
+ */
+
+#include "xzre.h"
+
+u8 *elf_find_string(elf_info_t *elf_info, EncodedStringId encoded_string_id, u8 *code_start) {
```

Comment:
this is incorrect. `encoded_string_id` is passed as an in-out pointer (look for the correct prototype in `xzre.h`)